### ISSUE TITLE
Audio over DS4 jack

### ISSUE THREAD

**Issue by dovahkiin98** [Ryochan7/DS4Windows#275]
Is there a guide on how to make the audio over the controller's jack working? I've tried USB, with DS4Windows, without DS4Windows, bluetooth, nothing seems to work.

This is the list of audio devices, it's detecting the controller with full volume, but when I run the test it's not working.
![image](https://user-images.githubusercontent.com/8161112/39420168-3077a726-4c18-11e8-9950-4d54d61dea4e.png)

Also, my headphones are 1 jack (audio + mic, 3 strips), does that change anything?

**Comment by Ryochan7** [comment]
I believe the 3 strip jack can be used fine. I have tried using a pair of headphones with a built-in mic and headphone and microphone support worked fine when connected to a DualShock 4. Headphone support will only work for the DS4 when connected via the Sony wireless dongle or when a DS4 v2 is connected via USB. There is no support to use the headphone jack on the DS4 when connected via Bluetooth.

Older versions of DS4Windows would disable the headphone volume when generating output reports. That change would not be visible in the Windows audio control panel. Those kinds of issues were mostly eliminated a while ago.

**Comment by mjhorvath** [comment]
Any plans to enable audio over Bluetooth? Is the Sony wireless dongle a good device to use as an alternative? Thanks.

**Comment by Ryochan7** [comment]
I don't plan on looking into whether getting audio support working over Bluetooth is possible. There has been some testing done with the ds4drv project but it didn't seem to evolve past the experimental stage. The Sony Wireless Adapter works pretty well and it is my preferred connection type lately. The device is locked at an 8 ms poll rate so it can't be as snappy as what is possible with a Bluetooth connection but the difference seems fairly minimal. The big problem that I have with the device is that it always presents a controller to Windows even when a DS4 is not synced with it. That can cause a problem if you use the hold PS button to disconnect method and the adapter continues to tell Windows that the PS button is still held down.

**Comment by mjhorvath** [comment]
Good to know, thanks!

**Comment by epigramx** [comment]
Nah, I don't think you should give up on that. A Sony adapter is an extra cost but also technologically not perfect, especially because of the extra delay.

**Comment by haywirephoenix** [comment]
With the adapter being overpriced (extortion by design if you need bt audio), and it also now being out of stock everywhere; this could prove to be a really interesting project. 

I have an EkoBuy Bluetooth 4.0 with BLE, EDR A2DP/HFP/HSP/AVRCP. It's CSR but it works fine with generic microsoft drivers.

I've heard some say that when pairing, it briefly advertises the audio device. And some people in russia have actally been able to send bt audio packets in linux https://habr.com/en/post/302356/

**Comment by haywirephoenix** [comment]
#1268
